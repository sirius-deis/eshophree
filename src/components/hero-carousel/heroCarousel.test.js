import { render, screen, fireEvent, act } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import HeroCarousel from './heroCarousel';

jest.useFakeTimers();

describe('HeroCarousel component', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <HeroCarousel />
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });
  it('should change slide when next control is clicked', () => {
    const { getByText } = render(<MemoryRouter><HeroCarousel /></MemoryRouter>);
    const nextControl = getByText('›');
    act(() => {
      fireEvent.click(nextControl);
    });
    expect(screen.getAllByAltText("slide")[0].parentElement).not.toHaveClass("active");
    expect(screen.getAllByAltText("slide")[1].parentElement).toHaveClass("active");
  });
  it('should change slide when previous control is clicked', () => {
    const { getByText } = render(<MemoryRouter><HeroCarousel /></MemoryRouter>);
    const prevControl = getByText('‹');
    act(() => {
      fireEvent.click(prevControl);
    });
    expect(screen.getAllByAltText("slide")[3].parentElement).toHaveClass("active");
    expect(screen.getAllByAltText("slide")[0].parentElement).not.toHaveClass("active");
  });
  it('automatically changes slide after interval', () => {
    const { getAllByAltText } = render(<MemoryRouter><HeroCarousel /></MemoryRouter>);
    act(() => {
      jest.advanceTimersByTime(10000);
    });
    expect(getAllByAltText("slide")[1].parentElement).toHaveClass("active");
    expect(getAllByAltText("slide")[0].parentElement).not.toHaveClass("active");
  });
  it('should change slide when indicator is clicked', () => {
    const { container } = render(<MemoryRouter><HeroCarousel /></MemoryRouter>);
    const indicators = container.querySelectorAll('button');
    act(() => {
      fireEvent.click(indicators[1]);
    });
    expect(indicators[1]).toHaveClass('active');
  });
});
