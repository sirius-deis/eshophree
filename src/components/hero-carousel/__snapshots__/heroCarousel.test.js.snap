// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroCarousel component should match snapshot 1`] = `
<div>
  <div
    class="sc-bgqQcB dxzDM"
  >
    <div
      class="sc-gxYJeL dmSiwI"
    >
      <button
        class="active"
      />
      <button
        class=""
      />
      <button
        class=""
      />
      <button
        class=""
      />
    </div>
    <div
      class="sc-gTRrQi fZNjYN"
      style="transform: translateX(0%);"
    >
      <div
        class="sc-ewnqHT hufTMZ"
      >
        <img
          alt="slide"
          src="https://source.unsplash.com/random/300×300?electronics"
        />
        <div
          class="sc-kPTPQs gkVKTH"
        >
          <button
            class="sc-ispOId sc-kuWgmH dHSeuY jjKghz"
          >
            <a
              href="/shop/"
            >
              Now available →
            </a>
          </button>
        </div>
      </div>
      <div
        class="sc-ewnqHT hufTMZ"
      >
        <img
          alt="slide"
          src="https://source.unsplash.com/random/300×300?electronics"
        />
        <div
          class="sc-kPTPQs gkVKTH"
        >
          <button
            class="sc-ispOId sc-kuWgmH dHSeuY jjKghz"
          >
            <a
              href="/shop/"
            >
              Now available →
            </a>
          </button>
        </div>
      </div>
      <div
        class="sc-ewnqHT hufTMZ"
      >
        <img
          alt="slide"
          src="https://source.unsplash.com/random/300×300?electronics"
        />
        <div
          class="sc-kPTPQs gkVKTH"
        >
          <button
            class="sc-ispOId sc-kuWgmH dHSeuY jjKghz"
          >
            <a
              href="/shop/"
            >
              Now available →
            </a>
          </button>
        </div>
      </div>
      <div
        class="sc-ewnqHT hufTMZ"
      >
        <img
          alt="slide"
          src="https://source.unsplash.com/random/300×300?electronics"
        />
        <div
          class="sc-kPTPQs gkVKTH"
        >
          <button
            class="sc-ispOId sc-kuWgmH dHSeuY jjKghz"
          >
            <a
              href="/shop/"
            >
              Now available →
            </a>
          </button>
        </div>
      </div>
    </div>
    <button
      class="sc-fFGjHI sc-dicizt hNdyPn hpXFoe"
    >
      ‹
    </button>
    <button
      class="sc-fFGjHI sc-iVCKna hNdyPn bbzcUU"
    >
      ›
    </button>
  </div>
</div>
`;
